pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -140,21 +126,14 @@ func NewManagerImpl(topology []cadvisorapi.Node, topologyAffinityStore topologym
 func newManagerImpl(socketPath string, topology []cadvisorapi.Node, topologyAffinityStore topologymanager.Store) (*ManagerImpl, error) {
 	klog.V(2).InfoS("Creating Device Plugin manager", "path", socketPath)
 
-	if socketPath == "" || !filepath.IsAbs(socketPath) {
-		return nil, fmt.Errorf(errBadSocket+" %s", socketPath)
-	}
-
```

Comment:
@klueska can you explain why this validation code was removed? Should it be moved to plugin.NewServer instead?

Reply:
In this case it looks like `manager.checkpointdir, _ = filepath.Split(socketPath)` is called before validation, correct?

Reply:
I was trying to avoid having `NewServer()` return an error along with the server object. While it does do the split / join before validation, the socket will never end up being used before it is validated. I can revisit if you feel strongly, but I don't see there being an actual issue at runtime doing things in this order.

Reply:
no strong feelings. If it works for you, i'm ok with it.

Reply:
I changed my mind and made the adjustment. Thanks for looking so carefully through the code.